ietf/doc/views_statement.py

```diff
@@ -17,4 +29,247 @@ def serve_pdf(self, name, rev=None):
+                    "application/pdf",
+                ) + settings.DOC_TEXT_FILE_VALID_UPLOAD_MIME_TYPES
+                if not content_type.startswith(
+                    acceptable_types
+                ):  # dances around decoration of types with encoding etc.
+                    self.add_error(
+                        "statement_file",
+                        forms.ValidationError(
+                            f"Unexpected content type: Expected one of {', '.join(acceptable_types)}"
+                        ),
+                    )
+                elif content_type != "application/pdf":
+                    markdown_content = get_cleaned_text_file_content(
+                        self.cleaned_data["statement_file"]
+                    )
+        if markdown_content != "":
+            try:
+                _ = markdown.markdown(markdown_content)
+            except Exception as e:
+                raise forms.ValidationError(f"Markdown processing failed: {e}")
+
+
+@role_required("Secretariat")
+def submit(request, name):
+    statement = get_object_or_404(Document, type="statement", name=name)
+
+    if request.method == "POST":
+        form = StatementUploadForm(request.POST, request.FILES)
+        if form.is_valid():
+            statement_submission = form.cleaned_data["statement_submission"]
+            writing_pdf = (
+                statement_submission == "upload"
+                and form.cleaned_data["statement_file"].content_type
+                == "application/pdf"
+            )
+
+            statement.rev = "%02d" % (int(statement.rev) + 1)
+            statement.uploaded_filename = (
+                f"{statement.name}-{statement.rev}.{'pdf' if writing_pdf else 'md'}"
+            )
+            e = NewRevisionDocEvent.objects.create(
+                type="new_revision",
+                doc=statement,
+                by=request.user.person,
+                rev=statement.rev,
+                desc="New revision available",
+            )
+            statement.save_with_history([e])
+            markdown_content = ""
+            if statement_submission == "upload":
+                if not writing_pdf:
+                    markdown_content = get_cleaned_text_file_content(
+                        form.cleaned_data["statement_file"]
+                    )
+            else:
+                markdown_content = form.cleaned_data["statement_content"]
+            with Path(statement.get_file_name()).open(
+                mode="wb" if writing_pdf else "w"
+            ) as destination:
+                if writing_pdf:
+                    for chunk in form.cleaned_data["statement_file"].chunks():
+                        destination.write(chunk)
+                else:
+                    destination.write(markdown_content)
+            return redirect("ietf.doc.views_doc.document_main", name=statement.name)
+
+    else:
+        if statement.uploaded_filename.endswith("pdf"):
```

Comment:
Might want to use something a little more robust, `Path(...).suffix.lower()` perhaps? (Not sure it's worth it, given that it'd still be identifying the format by its extension, so the endswith might be adequate.)

Reply:
again, its our code that set the uploaded_filename.